noir-projects/noir-protocol-circuits/crates/types/src/constants.nr

```diff
@@ -271,10 +271,12 @@ pub global PRIVATE_LOG_LENGTH: u32 = PRIVATE_LOG_SIZE_IN_FIELDS + 1 /* length */
 pub global PRIVATE_LOG_DATA_LENGTH: u32 = PRIVATE_LOG_LENGTH
     + 1 /* note_hash_counter */
     + 1 /* counter */;
-/// A private log's ciphertext is all fields except the first one, which is called 'tag'. We call this 'ciphertext'
-/// because, unlike public logs, it is expected that private logs are encrypted (though this is not
-/// mandatory).
-pub global PRIVATE_LOG_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_SIZE_IN_FIELDS - 1; // -1 for the tag
+/// Messages are either delivered via a private log or emitted as an offchain message. The ciphertext length is derived
+/// from private log related constants because we want the message length to be consistent no matter the delivery mode
+/// (this allows us to reuse message processing functionality, see `do_process_message` function).
+///
+/// A private log contains the message ciphertext and the tag. Hence we subtract 1 for the tag.
+pub global MESSAGE_CIPHERTEXT_LEN: u32 = PRIVATE_LOG_SIZE_IN_FIELDS - 1;
```

Comment:
Not a fan of having message length in protocol circuits. The private log ciphertext is fine because the tag _is_ a protocol concern (since nodes index by tag etc), but _our_ usage of logs for messages is an aztec-nr thing. So I'd keep `PRIVATE_LOG_CIPHERTEXT_LEN` in here, and have definitions for `MESSAGE_CIPHER_LEN` in our message mod.